pkg/spoke/controllers/statuscontroller/availablestatus_controller.go

```diff
@@ -102,13 +104,25 @@ func (c *AvailableStatusController) syncManifestWork(ctx context.Context, origin
 
 	needStatusUpdate := false
 	// handle status condition of manifests
+	// TODO revist this controller since this might bring races when user change the manifests in spec.
 	for index, manifest := range manifestWork.Status.ResourceStatus.Manifests {
-		availableStatusCondition := buildAvailableStatusCondition(manifest.ResourceMeta, c.spokeDynamicClient)
+		obj, availableStatusCondition := buildAvailableStatusCondition(manifest.ResourceMeta, c.spokeDynamicClient)
 		newConditions := helper.MergeStatusConditions(manifest.Conditions, []metav1.Condition{availableStatusCondition})
-		if !reflect.DeepEqual(manifestWork.Status.ResourceStatus.Manifests[index].Conditions, newConditions) {
+		if !equality.Semantic.DeepEqual(manifestWork.Status.ResourceStatus.Manifests[index].Conditions, newConditions) {
 			manifestWork.Status.ResourceStatus.Manifests[index].Conditions = newConditions
 			needStatusUpdate = true
 		}
+
+		values, err := c.getFeedbackValues(manifest.ResourceMeta, obj, manifestWork.Spec.ManifestConfigs)
```

Comment:
I gather that these logically distinct controllers are combined because of the cost of the live-get?  If this is the case, the reasoning should be commented for future readers.

Reply:
a comment on the reason is added at the controller definition.

Reply:
in the legacy kubefed we had a multi-cluster informer that dynamically watches resources by api discovery. if we can replace the live-read w/ informer-read, the cost will be significantly reduced.

Reply:
yeah, there is good/bad to have such informers. given the current informer filtering mechanism. We might monitor a lot of unrelated resource on the spoke. It would be quite costly on a large spoke I think.